src/TestPlan.lua

```diff
@@ -2,68 +2,259 @@
 	Represents a tree of tests that have been loaded but not necessarily
 	executed yet.
 
-	TestPlan objects are produced by TestPlanner and TestPlanBuilder.
+	TestPlan objects are produced by TestPlanner.
 ]]
 
 local TestEnum = require(script.Parent.TestEnum)
+local Expectation = require(script.Parent.Expectation)
 
-local TestPlan = {}
+local function newEnvironment(currentNode, extraEnvironment)
+	local env = {}
 
-TestPlan.__index = TestPlan
+	if extraEnvironment then
+		if type(extraEnvironment) ~= "table" then
+			error(("Bad argument #2 to newEnvironment. Expected table, got %s"):format(
+				typeof(extraEnvironment)), 2)
+		end
 
---[[
-	Create a new, empty TestPlan.
-]]
-function TestPlan.new()
-	local self = {
-		children = {}
+		for key, value in pairs(extraEnvironment) do
+			env[key] = value
+		end
+	end
+
+	function env.describeFOCUS(phrase, callback)
+		return env.describe(phrase, callback, TestEnum.NodeModifier.Focus)
+	end
+
+	function env.describeSKIP(phrase, callback)
+		return env.describe(phrase, callback, TestEnum.NodeModifier.Skip)
+	end
+
+	function env.describe(phrase, callback, nodeModifier)
+		local node = currentNode:addChild(phrase, TestEnum.NodeType.Describe, nodeModifier)
+		node.callback = callback
+		node:expand()
+		return node
```

Comment:
Just noticed that we return TestEZ's internal nodes from these functions, which might let users get ahold of internal state.

Do we use this anywhere else?

Reply:
Hmm, not exactly. itFIXME uses it's return, but I can probably refactor that.

Reply:
Done.